2.5/README.md

```diff
@@ -23,26 +23,111 @@ modules for their web applications. There is no guarantee for any specific npm o
 version, that is included in the image; those versions can be changed anytime and
 the nodejs itself is included just to make the npm work.
 
-Usage
+Usage in Openshift
 ---------------------
-For this, we will assume that you are using the `rhscl/ruby-25-rhel7 image`, available via `ruby:2.5` imagestream tag in Openshift.
+For this, we will assume that you are using the `ubi8/ruby-25 image`, available via `ruby:2.5` imagestream tag in Openshift.
 Building a simple [ruby-sample-app](https://github.com/sclorg/s2i-ruby-container/tree/master/2.5/test/puma-test-app) application
 in Openshift can be achieved with the following step:
 
     ```
-    oc new-app ruby:2.5~https://github.com/sclorg/s2i-ruby-container.git --context-dir=2.5/test/puma-test-app/
+    oc new-app ruby:2.5~https://github.com/sclorg/rails-ex.git
     ```
 
-The same application can also be built using the standalone [S2I](https://github.com/openshift/source-to-image) application on systems that have it available:
+**Accessing the application:**
+```
+$ oc get pods
+$ oc exec <pod> -- curl 127.0.0.1:8080
+```
 
-    ```
-    $ s2i build https://github.com/sclorg/s2i-ruby-container.git --context-dir=2.5/test/puma-test-app/ rhscl/ruby-25-rhel7 ruby-sample-app
-    ```
+Source-to-Image framework and scripts
+-------------------------------------
+This image supports the [Source-to-Image](https://docs.openshift.com/container-platform/4.4/builds/build-strategies.html#images-create-s2i_build-strategies)
+(S2I) strategy in OpenShift. The Source-to-Image is an OpenShift framework
+which makes it easy to write images that take application source code as
+an input, use a builder image like this Ruby container image, and produce
+a new image that runs the assembled application as an output.
 
-**Accessing the application:**
+To support the Source-to-Image framework, important scripts are included in the builder image:
+
+* The `/usr/libexec/s2i/assemble` script inside the image is run to produce a new image with the application artifacts. The script takes sources of a given application and places them into appropriate directories inside the image. It utilizes some common patterns in Ruby application development (see the **Environment variables** section below).
+* The `/usr/libexec/s2i/run` script is set as the default command in the resulting container image (the new image with the application artifacts). It configures `puma` server and runs a web server on port 8080 using `rackup`.
```

Comment:
This is incorrect. It runs either [puma](https://github.com/sclorg/s2i-ruby-container/blob/4ccde61cf6d69c9fd593db972422cc4d27c9cbe1/2.7/s2i/bin/run#L25) **OR** [rackup](https://github.com/sclorg/s2i-ruby-container/blob/4ccde61cf6d69c9fd593db972422cc4d27c9cbe1/2.7/s2i/bin/run#L33).

`puma` **IS**  a web server.

`rackup` is a generic mechanism to start a web app (usually in form of a [web server](https://github.com/rack/rack#supported-web-servers-)).

Those have different kind of configs / runtimes, and cannot be used interchangably. On top of that, `rackup` can be run both with and without `bundled` environment.